server-common/src/main/java/org/apache/kafka/admin/AdminUtils.java

```diff
@@ -166,7 +164,7 @@ private static Map<Integer, List<Integer>> assignReplicasToBrokersRackAware(int
                                                                                 int fixedStartIndex,
                                                                                 int startPartitionId) {
         Map<Integer, String> brokerRackMap = new HashMap<>();
-        brokerMetadatas.forEach(m -> brokerRackMap.put(m.id, m.rack.get()));
+        brokerMetadatas.forEach(m -> brokerRackMap.put(m.id, m.rack.orElseThrow(() -> new AdminOperationException("Not all brokers have rack information for replica rack aware assignment."))));
```

Comment:
Could you please add UT for it? `AdminRackAwareTest` is a good class to have the new test.

Reply:
> Could you please add UT for it? `AdminRackAwareTest` is a good class to have the new test.

`Sure, I'm on it`

Reply:
> Could you please add UT for it? `AdminRackAwareTest` is a good class to have the new test.

*I have added a new unit test for it. If there are any areas that could be improved, please provide me with some guidance and suggestions.*